pkg/assembler/backends/neo4j/artifact.go

```diff
@@ -55,30 +54,26 @@ func (c *neo4jClient) Artifacts(ctx context.Context, artifactSpec *model.Artifac
 	session := c.driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeRead})
 	defer session.Close()
 
-	if artifactSpec.Algorithm == nil && artifactSpec.Digest == nil {
-		return nil, gqlerror.Errorf("must specify both algorithm and digest for artifact")
-	}
+	var sb strings.Builder
+	var firstMatch bool = true
+	queryValues := map[string]any{}
 
-	result, err := session.ReadTransaction(
-		func(tx neo4j.Transaction) (interface{}, error) {
+	sb.WriteString("MATCH (n:Artifact)")
 
-			var sb strings.Builder
-			var firstMatch bool = true
-			queryValues := map[string]any{}
-
-			sb.WriteString("MATCH (n:Artifact)")
+	if artifactSpec.Algorithm != nil {
+		matchProperties(&sb, firstMatch, "n", "algorithm", "$artifactAlgo")
+		firstMatch = false
+		queryValues["artifactAlgo"] = strings.ToLower(*artifactSpec.Algorithm)
```

Comment:
Shouldn't the ingest be enforcing the lowercase? Reason why I didn't add it to the query

Reply:
Ingest does, but we also want the query side to do it too.

```gql
query ArtQ5 {
  artifacts(artifactSpec: {digest: "7A8F47318E4676DACB0142AFA0B83029CD7BEFD9"}) {
    ...allArtifactTree
  }
}
```

would return nothing otherwise. We need to handle it here otherwise in each place we would run the query we'd need to convert o lower case.

Reply:
oh wait disregard.